ideascube/forms.py

```diff
@@ -52,3 +60,37 @@ def save(self, *args, **kwargs):
         user.is_staff = True
         user.save()
         return user
+
+
+class UserImportForm(forms.Form):
+    source = forms.FileField(required=True)
+
+    def save(self):
+        source = self.cleaned_data['source']
+        users = []
+        errors = []
+        rows = csv.DictReader(source.read().decode().splitlines())
```

Comment:
What I don't like here is that your use of `read` means we take the whole file in memory. If the file is long, that could be a problem.

Working with an iterator would be IMHO better here, and that's why I suggested to pass the file-like object directly to `csv.DictReader`.

But you're right, that doesn't work because of `bytes` vs `str`.

In this case, you can do:

```python
from io import TextIOWrapper

source = TextIOWrapper(self.cleaned_data['source'].file, encoding='utf-8')
rows = csv.DictReader(source)
```

And that should consume the file line by line, without taking it all in memory. :smiley: